sympy/solvers/solvers.py

```diff
@@ -1860,40 +1860,35 @@ def _solve_system(exprs, symbols, **flags):
             result = [result] if result else []
             if failed:
                 if result:
-                    solved_syms = list(result[0].keys())  # there is only one result dict
+                    solved_syms = list(result[0].keys())
                 else:
                     solved_syms = []
             # linear doesn't change
         else:
             linear = False
-            if len(symbols) > len(polys):
+            lex_syms = sorted(symbols, key=lambda s: s.sort_key())
+            gb = groebner(polys, lex_syms)
+            independent_polys = [simplify(poly) for poly in gb if poly != 0]
+            rank = len(independent_polys)
 
-                free = set().union(*[p.free_symbols for p in polys])
-                free = list(ordered(free.intersection(symbols)))
-                got_s = set()
-                result = []
-                for syms in subsets(free, min(len(free), len(polys))):
-                    try:
-                        # returns [], None or list of tuples
-                        res = solve_poly_system(polys, *syms)
-                        if res:
-                            for r in set(res):
-                                skip = False
-                                for r1 in r:
-                                    if got_s and any(ss in r1.free_symbols
-                                           for ss in got_s):
-                                        # sol depends on previously
-                                        # solved symbols: discard it
-                                        skip = True
-                                if not skip:
-                                    got_s.update(syms)
-                                    result.append(dict(list(zip(syms, r))))
-                    except NotImplementedError:
-                        pass
-                if got_s:
-                    solved_syms = list(got_s)
+            if rank < len(symbols) and 1 not in gb:
+                final_solution = _solve_poly_system_underdetermined(gb, lex_syms)
```

Comment:
The number of polynomials does not work like the "rank" in the same way as for linear equations. There can be more equations than symbols but still an underdetermined system.